mlflow/types/utils.py

```diff
@@ -196,6 +237,12 @@ def _infer_schema(data: Any) -> Schema:
     #        ColSpec(type=Object([Property(name='key', dtype=DataType.string),
     #                             Property(name='key2', dtype=DataType.string, required=False)]
     #               ), name='dict')])
+
+    # To keep backward compatibility with 2.7.1, an empty list is inferred as string.
```

Comment:
```suggestion
    # To keep backward compatibility with < 2.9.0, an empty list is inferred as string.
```

(assuming we'll ship this in 2.9.0)